prow/plugins/lgtm/lgtm.go

```diff
@@ -83,82 +96,108 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent) e
 	// If we create an "/lgtm" comment, add lgtm if necessary.
 	// If we create a "/lgtm cancel" comment, remove lgtm if necessary.
 	wantLGTM := false
-	if lgtmRe.MatchString(e.Body) {
+	if lgtmRe.MatchString(body) {
 		wantLGTM = true
-	} else if lgtmCancelRe.MatchString(e.Body) {
+	} else if lgtmCancelRe.MatchString(body) {
 		wantLGTM = false
 	} else {
 		return nil
 	}
 
-	org := e.Repo.Owner.Login
-	repo := e.Repo.Name
-	commentAuthor := e.User.Login
+	// Author cannot LGTM own PR
+	isAuthor := author == issueAuthor
+	if isAuthor && wantLGTM {
+		resp := "you cannot LGTM your own PR."
+		log.Infof("Commenting with \"%s\".", resp)
+		return gc.CreateComment(repo.Owner.Login, repo.Name, number, plugins.FormatResponseRaw(body, htmlURL, author, resp))
+	}
+
+	return handle(wantLGTM, author, issueAuthor, repo, assignees, number, body, htmlURL, gc, log)
+}
 
-	// Allow authors to cancel LGTM. Do not allow authors to LGTM, and do not
-	// accept commands from any other user.
+func handlePullRequestReview(gc githubClient, log *logrus.Entry, e github.ReviewEvent) error {
+	author := e.Review.User.Login
+	issueAuthor := e.PullRequest.User.Login
+	repo := e.Repo
+	assignees := e.PullRequest.Assignees
+	number := e.PullRequest.Number
+	body := e.Review.Body
+	htmlURL := e.Review.HTMLURL
+
+	// If we review with Approve, add lgtm if necessary.
+	// If we review with Request Changes, remove lgtm if necessary.
+	wantLGTM := false
+	if e.Review.State == "approve" {
+		wantLGTM = true
+	} else if e.Review.State == "request_changes" {
```

Comment:
What happens if someone sets the review to "request_changes" but also comments `/lgtm` in the review body?

Reply:
I was wondering about that.

Does the generic comment handler listen to review body events? 

In general, I say buttons win over comment in this case. Reviewers should be familiar with the Github interface and process that allows folks to review.

Reply:
+1 to Cole's comment. I'm much less likely to accidentally type `/lgtm` than click the wrong review response bubble.